File: src/Web/SWARM.PuP.Web/Models/CurrentUserInfo.cs
﻿using System.Collections.Generic;
using SWARM.PuP.Web.Models;

namespace SWARM.PuP.Web.ViewModels
{
    public class CurrentUserInfo : UserInfo
    {
        public CurrentUserInfo(PuPUser user)
        {
            this.Id = user.Id;

            this.UserName = user.UserName;

            this.Email = user.Email;

            this.Tags = user.Tags;

            this.PortraitUrl = user.PortraitUrl;
        }

        public string Email { get; set; }
        
        public IEnumerable<PuPTag> Tags { get; set; }
    }

    public class CurrentUserToken : CurrentUserInfo
    {
        public CurrentUserToken(PuPUser user) : base(user)
        {

        }

        public string AccessToken { get; set; }

        public long ExpiresIn { get; set; }
    }
}

File: src/Android/app/src/main/java/com/swarmnyc/pup/module/models/CurrentUserInfo.java
package com.swarmnyc.pup.module.models;

import com.google.gson.annotations.SerializedName;

import java.util.ArrayList;
import java.util.HashSet;
import java.util.List;
import java.util.Set;

public class CurrentUserInfo extends UserInfo
{
	public static final CurrentUserInfo Null = new CurrentUserInfo();
	private String       email;
	private String       accessToken;
	private double       expiresIn;
	private List<PuPTag> tags;

	@SerializedName( "media" )
	private Set<String>  socialMedia;

	public CurrentUserInfo()
	{
		tags = new ArrayList<>();
		socialMedia = new HashSet<>();
	}

	public String getEmail()
	{
		return email;
	}

	public void setEmail( String email )
	{
		this.email = email;
	}

	public String getAccessToken()
	{
		return accessToken;
	}

	public void setAccessToken( final String accessToken )
	{
		this.accessToken = accessToken;
	}

	public double getExpiresIn()
	{
		return expiresIn;
	}

	public void setExpiresIn( double expiresIn )
	{
		this.expiresIn = expiresIn;
	}

	public List<PuPTag> getTags()
	{
		return tags;
	}

	public void setTags( List<PuPTag> tags )
	{
		this.tags = tags;
	}

	public String getTagValue( String key )
	{
		return getTagValue(key, null);
	}

	public String getTagValue( String key, String defaultValue )
	{
		for ( PuPTag tag : tags )
		{
			if ( tag.key.equals( key ) )
			{
				return tag.value;
			}
		}

		return defaultValue;
	}

	public Set<String> getSocialMedia()
	{
		return socialMedia;
	}

	public void setSocialMedia(Set<String> socialMedia)
	{
		this.socialMedia = socialMedia;
	}

	public boolean hasSocialMedium(String type)
	{
		for ( String m : socialMedia)
		{
			if ( type.equals( m ) )
			{ return true; }
		}

		return false;
	}
}
